cognite/client/data_classes/geospatial.py

```diff
@@ -0,0 +1,190 @@
+from typing import Any, Dict
+
+from cognite.client import utils
+from cognite.client.data_classes._base import CogniteResource, CogniteResourceList
+
+
+class FeatureType(CogniteResource):
+    """A representation of a feature type in the geospatial api."""
+
+    def __init__(
+        self,
+        external_id: str = None,
+        created_time: int = None,
+        last_updated_time: int = None,
+        properties: Dict[str, Any] = None,
+        search_spec: Dict[str, Any] = None,
+        cognite_client=None,
+    ):
+        self.external_id = external_id
+        self.created_time = created_time
+        self.last_updated_time = last_updated_time
+        self.properties = properties
+        self.search_spec = search_spec
+        self._cognite_client = cognite_client
+
+    @classmethod
+    def _load(cls, resource: Dict, cognite_client=None):
+        instance = cls(cognite_client=cognite_client)
+        for key, value in resource.items():
+            snake_case_key = utils._auxiliary.to_snake_case(key)
+            setattr(instance, snake_case_key, value)
+        return instance
+
+
+class FeatureTypeList(CogniteResourceList):
+    _RESOURCE = FeatureType
+    _ASSERT_CLASSES = False
+
+
+class PropertyAndSearchSpec:
+    """A representation of a feature type property and search spec."""
+
+    def __init__(
+        self, properties: Dict[str, Any] = None, search_spec: Dict[str, Any] = None,
+    ):
+        self.properties = properties
+        self.search_spec = search_spec
+
+
+class FeatureTypeUpdate:
+    """A representation of a feature type update in the geospatial api."""
+
+    def __init__(
+        self, external_id: str = None, add: PropertyAndSearchSpec = None, cognite_client=None
+    ):
+        self.external_id = external_id
+        self.add = add
+        self._cognite_client = cognite_client
+
+
+class FeatureTypeUpdateList:
+    _RESOURCE = FeatureTypeUpdate
+    _ASSERT_CLASSES = False
+
+
+class Feature(CogniteResource):
+    """A representation of a feature in the geospatial api."""
+
+    def __init__(
+        self, external_id: str = None, cognite_client=None, **properties,
+    ):
+        self.external_id = external_id
+        for key in properties:
+            setattr(self, key, properties[key])
+        self._cognite_client = cognite_client
+
+    @classmethod
+    def _load(cls, resource: Dict, cognite_client=None):
+        instance = cls(cognite_client=cognite_client)
+        for key, value in resource.items():
+            snake_case_key = utils._auxiliary.to_snake_case(key)
+            setattr(instance, snake_case_key, value)
+        return instance
+
+
+def _is_geometry_type(property_type: str):
+    return property_type in {"POINT"}
```

Comment:
should there be the other geometry types beside POINT?

Reply:
Good question. Not all geometry types are supported by shapely. But I have only had the chance to deal with POINT and GEOMETRYCOLLECTION. POINT is supported, not GEOMETRYCOLLECTION. We can extend later I think. Or remove from_geopandas method for now.

Reply:
updated to "POINT", "POLYGON", "MULTIPOLYGON", "GEOMETRYCOLLECTION"